sdk/storage/azure-storage-blob/azure/storage/blob/_shared/shared_access_signature.py

```diff
@@ -0,0 +1,368 @@
+            convert values to UTC. If a date is passed in without timezone info, it
+            is assumed to be UTC.
+        :type expiry: datetime or str
+        :param start:
+            The time at which the shared access signature becomes valid. If
+            omitted, start time for this call is assumed to be the time when the
+            storage service receives the request. Azure will always convert values
+            to UTC. If a date is passed in without timezone info, it is assumed to
+            be UTC.
+        :type start: datetime or str
+        :param str id:
+            A unique value up to 64 characters in length that correlates to a
+            stored access policy. To create a stored access policy, use
+            set_blob_service_properties.
+        :param str ip:
+            Specifies an IP address or a range of IP addresses from which to accept requests.
+            If the IP address from which the request originates does not match the IP address
+            or address range specified on the SAS token, the request is not authenticated.
+            For example, specifying sip=168.1.5.65 or sip=168.1.5.60-168.1.5.70 on the SAS
+            restricts the request to those IP addresses.
+        :param str protocol:
+            Specifies the protocol permitted for a request made. The default value
+            is https,http. See :class:`~azure.storage.common.models.Protocol` for possible values.
+        :param str cache_control:
+            Response header value for Cache-Control when resource is accessed
+            using this shared access signature.
+        :param str content_disposition:
+            Response header value for Content-Disposition when resource is accessed
+            using this shared access signature.
+        :param str content_encoding:
+            Response header value for Content-Encoding when resource is accessed
+            using this shared access signature.
+        :param str content_language:
+            Response header value for Content-Language when resource is accessed
+            using this shared access signature.
+        :param str content_type:
+            Response header value for Content-Type when resource is accessed
+            using this shared access signature.
+        '''
+        resource_path = container_name + '/' + blob_name
+
+        sas = _SharedAccessHelper()
+        sas.add_base(permission, expiry, start, ip, protocol, self.x_ms_version)
+        sas.add_id(policy_id)
+        sas.add_resource('b')
+        sas.add_override_response_headers(cache_control, content_disposition,
+                                          content_encoding, content_language,
+                                          content_type)
+        sas.add_resource_signature(self.account_name, self.account_key, 'blob', resource_path)
+
+        return sas.get_token()
+
+    def generate_container(self, container_name, permission=None, expiry=None,
```

Comment:
also here, do we want to rename it?

Reply:
Thanks! These are private/internal methods, copied from the track1 SDK. I will look at refactoring them/rename as needed for a subsequent preview!